dials_scratch build broken

From a fresh bootstrap build on Ubuntu 18.04.2 LTS, after
```
libtbx.configure dials_scratch/
```
The build fails, where the relevant error message appears to be
```
In file included from /home/fcx32934/sw/cctbx/modules/dials_scratch/idy/algorithms/spot_prediction/reflection_predictor.cc:13:
/home/fcx32934/sw/cctbx/modules/dials_scratch/idy/algorithms/spot_prediction/reflection_predictor.h: In constructor 'dials::algorithms::StillsExperimentalReflectionPredictor::StillsExperimentalReflectionPredictor(boost::shared_ptr<dxtbx::model::BeamBase>, const dxtbx::model::Detector&, scitbx::mat3<double>, const cctbx::uctbx::unit_cell&, const cctbx::sgtbx::space_group_type&, const double&, const double&)':
/home/fcx32934/sw/cctbx/modules/dials_scratch/idy/algorithms/spot_prediction/reflection_predictor.h:40:43: error: no matching function for call to 'dials::algorithms::StillsDeltaPsiReflectionPredictor::StillsDeltaPsiReflectionPredictor(const boost::shared_ptr<dxtbx::model::BeamBase>&, const dxtbx::model::Detector&, scitbx::mat3<double>&, const cctbx::uctbx::unit_cell&, const cctbx::sgtbx::space_group_type&, const double&)'
   40 |         ci_predict_ray_(beam->get_s0(),mos) {}
      |                                           ^
In file included from /home/fcx32934/sw/cctbx/modules/dials_scratch/idy/algorithms/spot_prediction/reflection_predictor.h:19,
```

--- a/idy/algorithms/spot_prediction/reflection_predictor.cc
+++ b/idy/algorithms/spot_prediction/reflection_predictor.cc
@@ -8,6 +8,7 @@
  *  This code is distributed under the BSD license, a copy of which is
  *  included in the root directory of this package.
  */
+#include <memory>
 #include <boost/python.hpp>
 #include <boost/python/def.hpp>
 #include <dials_scratch/idy/algorithms/spot_prediction/reflection_predictor.h>
@@ -37,7 +38,7 @@ namespace dials { namespace algorithms { namespace boost_python {
 
     class_<Predictor>("StillsExperimentalReflectionPredictor", no_init)
       .def(init<
-          const boost::shared_ptr<BeamBase>,
+          const std::shared_ptr<BeamBase>,
           const Detector&,
           mat3<double>,
           const cctbx::uctbx::unit_cell&,

--- a/idy/algorithms/spot_prediction/reflection_predictor.h
+++ b/idy/algorithms/spot_prediction/reflection_predictor.h
@@ -10,6 +10,7 @@
 #define IDY_ALGORITHMS_SPOT_PREDICTION_REFLECTION_PREDICTOR_H
 
 #include <algorithm>
+#include <memory>
 #include <scitbx/constants.h>
 #include <dxtbx/model/beam.h>
 #include <dxtbx/model/detector.h>
@@ -28,7 +29,7 @@ namespace dials { namespace algorithms {
   public:
 
     StillsExperimentalReflectionPredictor(
-        const boost::shared_ptr<BeamBase> beam,
+        const std::shared_ptr<BeamBase> beam,
         const Detector &detector,
         mat3<double> ub,
         const cctbx::uctbx::unit_cell &unit_cell,

